plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizardModel.java

```diff
@@ -81,7 +81,6 @@
 	private boolean originatedFromDockerExplorer;
 	
 	public DeployImageWizardModel() {
-		setConnection(ConnectionsRegistrySingleton.getInstance().getRecentConnection(Connection.class));
```

Comment:
This is what was causing the problem.  It returns the one recently accessed but we want the one you selected.  So I defered it to being set in the handler. Hmm... maybe we need to explicitly set 'recent' when calling from the docker explorer side?